src/openMVG/localization/CCTagLocalizer.cpp

```diff
@@ -84,98 +77,97 @@ CCTagLocalizer::CCTagLocalizer(const std::string &sfmFilePath,
-  std::vector<int> counterCCtagsInImage = {0, 0, 0, 0, 0, 0};
-  // Read for each view the corresponding regions and store them
-  for(const auto &iter : sfm_data.GetViews())
+
+  // Read for each view the corresponding Regions and store them
+  for(const auto &iter : _sfm_data.GetViews())
   {
     const IndexT id_view = iter.second->id_view;
-    Reconstructed_RegionsCCTag& reconstructedRegion = _regions_per_view[id_view];
-
-    const std::string &sImageName = iter.second.get()->s_Img_path;
-    std::string featFilepath = stlplus::create_filespec(feat_directory, std::to_string(iter.first), ".feat");
-    std::string descFilepath = stlplus::create_filespec(feat_directory, std::to_string(iter.first), ".desc");
-
-    if(!(stlplus::is_file(featFilepath) && stlplus::is_file(descFilepath)))
+    if(observationsPerView.count(id_view) == 0)
+      continue;
+    const auto& observations = observationsPerView.at(id_view);
     {
-      // legacy compatibility, if the features are not named using the UID convention
-      // let's try with the old-fashion naming convention
-      const std::string basename = stlplus::basename_part(sImageName);
-      featFilepath = stlplus::create_filespec(feat_directory, basename, ".feat");
-      descFilepath = stlplus::create_filespec(feat_directory, basename, ".desc");
-      if(!(stlplus::is_file(featFilepath) && stlplus::is_file(descFilepath)))
+      const features::EImageDescriberType descType = _imageDescriber.getDescriberType();
+
+      if(observations.count(descType) == 0)
       {
-        OPENMVG_CERR("Cannot find the features for image " << sImageName 
-                << " neither using the UID naming convention nor the image name based convention");
-        return false;
+        // no descriptor of this type reconstructed in this View
+        _imageDescriber.Allocate(_regionsPerView.getData()[id_view][descType]);
+        continue;
       }
-    }
 
-    if(!reconstructedRegion._regions.Load(featFilepath, descFilepath))
-    {
-      OPENMVG_CERR("Invalid regions files for the view: " << sImageName);
-      return false;
+      // Load from files
+      std::unique_ptr<features::Regions> currRegions = sfm::loadRegions(feat_directory, id_view, _imageDescriber);
+
+      // Filter descriptors to keep only the 3D reconstructed points
+      _regionsPerView.getData()[id_view][descType] = createFilteredRegions(*currRegions, observations.at(descType), _reconstructedRegionsMappingPerView[id_view][descType]);
     }
-    
-    // Filter descriptors to keep only the 3D reconstructed points
-    reconstructedRegion.filterCCTagRegions(observationsPerView[id_view]);
-    
-    // Update the visibility mask
-    reconstructedRegion.updateLandmarksVisibility(presentIds);
-    
-    ++my_progress_bar;
   }
 
+
   {
+    std::set<int> presentCCtagIds;
+    std::vector<int> counterCCtagsInImage = {0, 0, 0, 0, 0, 0};
```

Comment:
ok, this is nitpicking and not related to the scope of the PR, but i don't remember why we stopped at 5 with the histogram. It would be better to use a std::map and do a real histogram. But ok it's just debugging stuff